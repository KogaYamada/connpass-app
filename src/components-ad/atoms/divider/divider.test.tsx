import React from 'react';
import Divider from './divider.component';
import { shallow, configure, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('divider snapshot tests', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  beforeEach(() => {
    wrapper = shallow(<Divider />);
  });

  it('<Divider />: propsなし', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('<Divider height="large"/>: サイズlarge', () => {
    wrapper.setProps({ height: 'large' })
    expect(wrapper).toMatchSnapshot();
  });

  it('<Divider height="middle"/>: サイズmiddle', () => {
    wrapper.setProps({ height: 'middle' })
    expect(wrapper).toMatchSnapshot();
  });

  it('<Divider height="small"/>: サイズsmall', () => {
    wrapper.setProps({ height: 'small' })
    expect(wrapper).toMatchSnapshot();
  });
});
