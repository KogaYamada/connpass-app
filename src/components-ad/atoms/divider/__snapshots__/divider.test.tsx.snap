// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divider snapshot tests <Divider />: propsなし 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "textAlign": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#707070",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`divider snapshot tests <Divider height="large"/>: サイズlarge 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "textAlign": "center",
        "width": "100%",
      },
      Object {
        "height": 100,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#707070",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`divider snapshot tests <Divider height="middle"/>: サイズmiddle 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "textAlign": "center",
        "width": "100%",
      },
      Object {
        "height": 50,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#707070",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`divider snapshot tests <Divider height="small"/>: サイズsmall 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "textAlign": "center",
        "width": "100%",
      },
      Object {
        "height": 20,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#707070",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;
